File: 403/python/2.py
T = input().strip()
U = input().strip()

found = False
for i in range(len(T) - len(U) + 1):
  flag = True
  for k in range(len(U)):
    if not (T[i+k] == U[k] or T[i+k] == '?'):
      flag = False
      break
  if flag:
    print("Yes")
    found = True
    break

if not found:
  print("No")

File: 403/2.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
  string T,U;
  bool flag;
  cin >> T >> U;

  for (int i=0;i<T.length();i++){
    flag = true;
    for (int k=0;k<U.length();k++){
      if (T[i+k] != U[k] && T[i+k] != '?'){
        flag = false;
        break;
      }
    }
    if (flag){
      cout << "Yes" << endl;
      return 0;
    } 
  }
  cout << "No" << endl;
  return 0;
}